corehq/apps/userreports/views.py

```diff
@@ -1216,18 +1219,32 @@ def data_source_status(request, domain, config_id):
     return json_response({'isBuilt': config.meta.build.finished})
 
 
-@login_and_domain_required
-def choice_list_api(request, domain, report_id, filter_id):
+@quickcache(['domain', 'report_id', 'filter_id'], timeout=5 * 60)
+def _get_report_filter(domain, report_id, filter_id):
     report = get_report_config_or_404(report_id, domain)[0]
     report_filter = report.get_ui_filter(filter_id)
     if report_filter is None:
         raise Http404(_(u'Filter {} not found!').format(filter_id))
+    return report_filter
 
+
+def _is_location_safe_choice_list(view_fn, domain, report_id, filter_id, **view_kwargs):
+    report_filter = _get_report_filter(domain, report_id, filter_id)
+    if hasattr(report_filter, 'choice_provider'):
+        return report_filter.choice_provider.location_safe
+    return False
```

Comment:
If you have a report with two choice list filters, one for locations (making the report location safe) and another for something else, does this mean the second filter will be inaccessible to restricted users?